apps/web-giddh/src/app/accounting/journal-voucher/invoice/invoice.component.html

```diff
@@ -0,0 +1,231 @@
+                                                (ngModelChange)="changePrice(i,$event)" (focus)="selectRow(true,i,null)"
+                                                [ngClass]="{'focus': isSelectedRow && selectedRowIdx === i}"
+                                                decimalDigitsDirective [DecimalPlaces]="2"
+                                                [disabled]="!transaction.inventory.stock.name" />
+                                        </td>
+                                        <td>
+                                            <input type="text" class="form-control text-left"
+                                                [(ngModel)]="transaction.inventory.unit.stockUnitCode" [readonly]="true"
+                                                (focus)="selectRow(true,i)"
+                                                [disabled]="!transaction.inventory.stock.name" />
+                                        </td>
+                                        <td>
+                                            <input onReturn type="text" class="form-control text-right" #toAmountField
+                                                id="stock_{{i}}" [(ngModel)]="transaction.inventory.amount"
+                                                (ngModelChange)="amountChanged(i)"
+                                                (focus)="selectRow(true,i);addNewStock(transaction.inventory.amount,transaction, i)"
+                                                (keydown.Tab)="addNewStock(transaction.inventory.amount,transaction, i)"
+                                                [ngClass]="{'focus': isSelectedRow && selectedRowIdx === i}"
+                                                decimalDigitsDirective [DecimalPlaces]="2"
+                                                [disabled]="!transaction.inventory.stock.name"
+                                                (blur)="checkIfEnteredAmountIsZero(transaction.inventory.amount, i, 'stock')" />
+                                        </td>
+                                    </tr>
+                                </ng-container>
+                            </ng-container>
+                            <tr *ngIf="stockTotal">
+                                <td colspan="4"></td>
+                                <td class>
+                                    <input type="text" [value]="stockTotal" readonly disabled
+                                        class="form-control text-right" />
+                                    <div class="bdrB"></div>
+                                </td>
+                            </tr>
+                            <br />
+                            <!-- accounts entry -->
+                            <!--  -->
+                            <ng-container *ngFor="let transaction of accountsTransaction;let i=index;let l=last">
+                                <tr [ngClass]="{'activeRow': isSelectedRow && selectedAccIdx === i}">
```

Comment:
rename index 'i'

Reply:
@rohit-walkover This code is copy-paste and work will be done in near future on this